### PR TITLE
Remove qutrit ops use of in-place inversion

### PR DESCRIPTION
**Pull request by albi3ro**
We are starting the process of getting rid of in-place inversion.  See PR #3518 .

Given the complexity of trying to remove such an intangled feature, I'm breaking removal of in-place inversion up into multiple smaller pieces. This first piece removes in-place inversion from the qutrit operations.

Instead of `TShift(0).inv()`, we should instead use `Adjoint(TShift(0))` or `Pow(TShift(0), -1)`.

@mudit2812

### CODE REVIEW

File: pennylane/ops/qutrit/non_parametric_ops.py [hunk 1]
```diff
@@ -565,19 +531,15 @@ def compute_matrix(subspace=None):  # pylint: disable=arguments-differ
...
 
     def pow(self, z):
-        if self.subspace is not None:
-            if not isinstance(z, int):
-                return super().pow(z)
-
-            return super().pow(z % 2)
-
-        return super().pow(z)
+        new_exp = z if self.subspace is None else z % 2
```
[the remark this answers is not in the document]
Comment (glassnotes) [reply pennylane/ops/qutrit/non_parametric_ops.py:544]: Ah, I think this would this warrant an additional test being added for period 4 things (that's beyond the scope of this PR, though)

**Review by glassnotes — COMMENTED** [review]
Thanks @albi3ro ! I left a couple questions about how this works.

File: pennylane/ops/qutrit/non_parametric_ops.py [hunk 2]
```diff
@@ -565,19 +531,15 @@ def compute_matrix(subspace=None):  # pylint: disable=arguments-differ
...
+    @property
+    def has_adjoint(self):  # pylint: disable=arguments-renamed, invalid-overridden-method
+        return self.subspace is not None
 
+    def adjoint(self):
         if self.subspace is None:
-            op.inverse = not self.inverse
-
-        return op
+            raise qml.operation.AdjointUndefinedError
```
[the remark this answers is not in the document]
Comment (albi3ro) [reply pennylane/ops/qutrit/non_parametric_ops.py:540]: You can think of `adjoint` and `pow` as being part of the decomposition flow.  If we have an adjoint, what do we decompose it to? If we have a power, what do we decompose it to?

So we always want `adjoint` and `pow` to return things that are more likely to be supported. We need a direction to prevent circular loops in the decomposition graph.

So we could *choose* to make adjoint more "fundamental" than power and allow power operations to decompose to adjoint, or we could choose to make power operations more fundamental and decompose adjoint operations to powers. But it's very important we don't do both.

Adjoints are more efficient, but powers are easy to combine. So far we have kept them separate, but I'm considering finally allowing powers to decompose to adjoints.

[the remark this answers is not in the document]
Comment (glassnotes) [reply pennylane/ops/qutrit/non_parametric_ops.py:540]: > So we could choose to make adjoint more "fundamental" than power and allow power operations to decompose to adjoint, or we could choose to make power operations more fundamental and decompose adjoint operations to powers. But it's very important we don't do both.

Oh I see, thanks for the additional context! For most of the basis gates I can think of, their adjoints can be expressed as some power of them (even if that power is 1 or -1), so expressing everything as a power makes a lot of sense.

[the remark this answers is not in the document]
Comment (albi3ro) [reply pennylane/ops/qutrit/non_parametric_ops.py:540]: I think choosing between power and adjoint as the more "fundamental" will take more discussion and thought. For the near future, we should leave it as if. If a user requests the power, then they want the power. If a user requests the adjoint, then they want the adjoint. We don't mix the two for them.

In a later PR, we can start to mix the two, but we would want to do so across PennyLane consistently.

File: pennylane/ops/qutrit/non_parametric_ops.py [hunk 3]
```diff
@@ -102,17 +102,7 @@ def compute_eigvals():
     # TODO: Add compute_decomposition once parametric ops are added.
 
     def pow(self, z):
-        if isinstance(z, int):
-            z_mod3 = z % 3
-            if z_mod3 < 2:
-                return super().pow(z_mod3)
-            return [self.adjoint()]
-        return super().pow(z)
```
Comment (AlbertMitjans) [inline pennylane/ops/qutrit/non_parametric_ops.py:110]: Why did you remove this logic?

**Review by glassnotes — APPROVED** [review]
Thanks for the clarifications on this @albi3ro !

File: doc/releases/changelog-dev.md [hunk 4]
```diff
@@ -150,6 +150,11 @@
 * Limit the `numpy` version to `<1.24`.
   [(#3563)](https://github.com/PennyLaneAI/pennylane/pull/3563)
 
+
+* Removes qutrit operations use of in-place inversion in preparation for the
+  removal of in-place inversion.
+  [(#3566)](https://github.com/PennyLaneAI/pennylane/pull/3566)
+
```
Comment (trbromley) [inline doc/releases/changelog-dev.md:157]: Could this be considered a breaking change?
